ksqldb-functional-tests/src/test/resources/query-validation-tests/fk-join.json

```diff
@@ -155,6 +251,25 @@
         "message": "Invalid join condition: table-table joins require to join on the primary key of the right input table. Got LEFT_TABLE.ID1 = RIGHT_TABLE.F2."
       }
     },
+    {
+      "name": "Should fail on right non-key attribute for inner-join (reverse join condition order) - feature flag enabled",
+      "properties": {
+        "ksql.joins.foreign.key.enable": true
+      },
+      "comments": [
+        "TODO: this test should be deleted once the feature is ungated and the feature flag is removed",
+        "as it duplicates another test above (without the feature flag)"
+      ],
+      "statements": [
+        "CREATE TABLE left_table (id1 BIGINT PRIMARY KEY, f1 BIGINT) WITH (kafka_topic='left_topic', format='JSON');",
+        "CREATE TABLE right_table (id2 BIGINT PRIMARY KEY, f2 BIGINT) WITH (kafka_topic='right_topic', format='JSON');",
+        "CREATE TABLE output AS SELECT id1, id2, f1, f2 FROM left_table JOIN right_table ON f2 = id1;"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "Invalid join condition: table-table joins require to join on the primary key of the right input table. Got LEFT_TABLE.ID1 = RIGHT_TABLE.F2."
```

Comment:
I think it'd be marginally better if the error message showed the join expressions in the same order as in the user's query but given the structure of the code right now this is somewhat difficult to achieve so I think this is fine.

Reply:
I had a similar though... But I think we could fix it: JoinTree could set a flag "flipped" that we preserve that allows us to flip back in the error message. Let me add this.